pkgs/shells/zsh/zsh-bd/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub}:
+
+stdenv.mkDerivation rec {
+  pname = "zsh-bd";
+  version = "2018-07-04";
+
+  src = fetchFromGitHub {
+    owner = "Tarrasch";
+    repo = pname;
+    rev = "d4a55e661b4c9ef6ae4568c6abeff48bdf1b1af7";
+    sha256 = "020f8nq86g96cps64hwrskppbh2dapfw2m9np1qbs5pgh16z4fcb";
+  };
+
+  phases = ["installPhase"];
+
+  installPhase = ''
+    mkdir -p $out/share/zsh-bd
+    cp {${src},$out/share/zsh-bd}/bd.zsh
```

Comment:
Then you can also do `cp {.,$out/share/zsh-bd}/bd.zsh` here (untested), since the source is unpacked into the current directory